### PR TITLE
br: fix br domain start

### LINKED ISSUE #55039: br block on create policy
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

br restore requires tidb Domain started to do ddls. after https://github.com/pingcap/tidb/pull/54913 we have separate domain Init and Start, br should start it explicitly.

### 2. What did you expect to see? (Required)
success
### 3. What did you see instead (Required)
block on running ddl
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #55039

Problem Summary:

### What changed and how does it work?
after https://github.com/pingcap/tidb/pull/54913, we separate domain Init and Start. BR also depends on domain and should start it to do DDLs
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by wuhuizuo** [comment]
/retest

@BornChanger br integration test timeouted.

**Review by lance6716 — APPROVED** [review]

File: br/pkg/gluetidb/glue_test.go [hunk 1]
```diff
@@ -20,21 +20,37 @@ import (
...
 func TestTheSessionIsoation(t *testing.T) {
 	req := require.New(t)
-	store := testkit.CreateMockStore(t)
+	store, dom := session.CreateStoreAndBootstrap(t)
 	ctx := context.Background()
 
-	g := Glue{}
-	session, err := g.CreateSession(store)
+	// we want to test glue start domain explicitly, so close it first.
+	dom.Close()
```
Comment (BornChanger) [inline br/pkg/gluetidb/glue_test.go:35]: is there case covering the case where dom is opened already?

Comment (D3Hunter) [reply br/pkg/gluetidb/glue_test.go:35]: check glue.GetDomain, it will get session twice, domain has started when create the second one

**Review by 3pointer — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**